lego/apps/notifications/notifications.py

```diff
@@ -21,9 +21,12 @@ def generate_mail(self):
             context={
                 "name": self.user.full_name,
                 "created_by": announcement.created_by.full_name,
+                "from_group": announcement.from_group.name
+                if announcement.from_group is not None
+                else None,
                 "message": announcement.message,
             },
-            subject=f"Viktig melding fra {announcement.created_by.full_name}",
+            subject=f"Viktig melding fra Abakus",
```

Comment:
Doesn't it make sense to put the group name here if the announcement is from a group? e.g "Viktig melding fra Arrkom"

Reply:
Just me being lazy, but yeah thats better.

Reply:
Thought I was supposed to just delete the old migrations and create new ones when I changed the models, should I not have replaced the initial one? Whats the correct procedure?

Reply:
When you change the models, you do `makemigrations` which creates a file that tells the database what changed and how it should adapt to it. All the migrations are connected, from the initial to the last, and the order of them are important. Deleting the old migrations and creating a new one only ever works if you can delete the database and start over from scratch, but that is not really an option in production.

So when you change the models, do `makemigrations`, `migrate` and push the new migration file along with the changed models. That way you don't break anything.

Reply:
It's not really optimized to have a bunch of migration files for every little change in the models. Sigrid is right that you can't delete the old migration file, but you can squash them.

### How to squach migrations in Django ?
[https://www.techiediaries.com/squash-django-migrations/](https://www.techiediaries.com/squash-django-migrations/)
Django has a management command that allows you to squach or combine multiple migration files into a single optimized migration file .

You can simply invoke this command using :
```
python manage.py squachmigrations 
Now lets consider this senario :
```
You have a Django app named myApp which has 4 migrations
```
- 0001_initial
- 0002_change_2
- 0003_change_3
- 0004_change_4
```
Now when you want to migrate your database ,applying these migrations takes time so you have decided to squach and optimize them .
```
manage.py squashmigrations myApp 0004
```
You pass the app name and migration file name (or just prefix) you want to squach up to .

You should get something like

Will squash the following migrations:
```
- 0001_initial
- 0002_change_2
- 0003_change_3
- 0004_change_4 
Do you wish to proceed? [yN] y
Optimizing...
Optimized from 10 operations to 5 operations.
Created new squashed migration /migrations/0001_squashed_0004_change_4.py 
```
After making sure everything is OK you can simply remove the old migration files and keep the squashed migration file alone .

As you can see from this simple example squashing is not only about combining migration files into one single file but also about optimizing the number of operations inside the final squashed file .

Reply:
I don’t think you should squash migrations if one of them is already applied on the server, since you will remove the migration that is the “current state” of the database.